### PR TITLE
Re-organized OpenStack documentation

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

**Comment by tzumainn** [comment]
@tomassedovic This does not yet contain any content updates; just a re-organization to see if it works for you. Besides the additions we talked about, I have the following comments. Let me know what you think!

* can we remove flannel references?
* do we actually provide load balancer servers? pretty sure no
* access control - do we link to anything useful?
* get rid of 'TODOs' mentioned in README.md under 'Advanced Configuration'?
* manage_packages variable - doesn't exist in openshift-ansible?
* break off post-install section into its own document?
* DNS and cinder sections are untouched, may benefit from a cleaning
* openshift_master_default_subdomain - not sure what the doc is trying to say

**Comment by bogdando** [comment]
Coud you please also verify this doesn't break external references given in https://github.com/openshift/openshift-docs/pull/6700 and https://github.com/openshift/openshift-docs/pull/6020 ?

**Comment by tzumainn** [comment]
@bogdando As far as I can tell this change should not affect that documentation, so I think we're good!

File: playbooks/openstack/README.md [hunk 1]
```diff
@@ -6,122 +6,97 @@ etc.). The result is an environment ready for OpenShift installation
...
+* Heat (Orchestration) must be available
+* The deployment image (CentOS 7.4 or RHEL 7) must be loaded
+* The deployment flavor must be available to your user
+  - `m1.medium` / 4GB RAM + 40GB disk should be enough for testing
+  - look at
+    the [Minimum Hardware Requirements page][hardware-requirements]
+    for production
+* The keypair for SSH must be available in OpenStack
+* `keystonerc` file that lets you talk to the OpenStack services
+   * NOTE: only Keystone V2 is currently supported
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/README.md]: Fixed!

File: playbooks/openstack/README.md [hunk 2]
```diff
@@ -6,122 +6,97 @@ etc.). The result is an environment ready for OpenShift installation
...
 
+You may also optionally want:
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/README.md]: Changed to 'Strongly recommended'

File: playbooks/openstack/README.md [hunk 3]
```diff
@@ -202,52 +177,265 @@ $ ansible-playbook --user openshift \
...
+
+This playbook is also located in the [custom-actions](https://github.com/openshift/openshift-ansible-contrib/blob/master/playbooks/provisioning/openstack/custom-actions) directory.
+It copies passed CAs to the trust chain location and updates the trust chain on each selected host.
+
+Example usage:
+```
+ansible-playbook -i <inventory> openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions/add-cas.yml --extra-vars '{"ca_files": [<absolute path to ca1 file>, <absolute path to ca2 file>]}'
+```
+
+Please consider contributing your custom playbook back to openshift-ansible-contrib!
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/README.md]: Fixed. Should we move the custom playbooks that are all still in openshift-ansible-contrib over here at some point?

File: playbooks/openstack/README.md [hunk 4]
```diff
@@ -202,52 +177,265 @@ $ ansible-playbook --user openshift \
...
 
+    $ ansible -i inventory masters[0] -m fetch -a "src=/bin/oc dest=oc"
+
+Once you obtain the `oc` binary, remember to put it in your `PATH`.
+
+
+### Logging in Using the Command Line
+
+```
+oc login --insecure-skip-tls-verify=true https://master-0.openshift.example.com:8443 -u user -p password
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/README.md]: Clarified!

File: playbooks/openstack/README.md [hunk 5]
```diff
@@ -202,52 +177,265 @@ $ ansible-playbook --user openshift \
...
+
+If you'd like to limit the run to one particular host, you can do so as follows:
+
+```
+ansible-playbook --private-key ~/.ssh/openshift -i inventory/ openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions/custom-playbook.yml -l app-node-0.openshift.example.com
+```
+
+You can also create your own custom playbook. Here are a few examples:
+
+#### Adding additional YUM repositories
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/README.md]: 100% agree! Just wanted to make sure people were okay with the configuration doc structure before doing something similar with the post-install stuff.

File: playbooks/openstack/README.md [hunk 6]
```diff
@@ -202,52 +177,265 @@ $ ansible-playbook --user openshift \
...
+
+A library of custom post-provision actions exists in `openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions`. Playbooks include:
+
+* [add-yum-repos.yml](https://github.com/openshift/openshift-ansible-contrib/blob/master/playbooks/provisioning/openstack/custom-actions/add-yum-repos.yml): adds a list of custom yum repositories to every node in the cluster
+* [add-rhn-pools.yml](https://github.com/openshift/openshift-ansible-contrib/blob/master/playbooks/provisioning/openstack/custom-actions/add-rhn-pools.yml): attaches a list of additional RHN pools to every node in the cluster
+* [add-docker-registry.yml](https://github.com/openshift/openshift-ansible-contrib/blob/master/playbooks/provisioning/openstack/custom-actions/add-docker-registry.yml): adds a list of docker registries to the docker configuration on every node in the cluster
+* [add-cas.yml](https://github.com/openshift/openshift-ansible-contrib/blob/master/playbooks/provisioning/openstack/custom-actions/add-rhn-pools.yml): adds a list of CAs to the trust chain on every node in the cluster
+
+
+### Scale Deployment up/down
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/README.md]: Ah, makes sense - removed.

File: playbooks/openstack/configuration.md [hunk 7]
```diff
@@ -0,0 +1,463 @@
...
+* `openshift_openstack_default_flavor` OpenStack flavor used by all VMs, unless a particular role flavor name is specified.
+* `openshift_openstack_master_flavor`
+* `openshift_openstack_infra_flavor`
+* `openshift_openstack_cns_flavor`
+* `openshift_openstack_node_flavor`
+* `openshift_openstack_lb_flavor`
+* `openshift_openstack_etcd_flavor`
+
+* `openshift_openstack_external_network_name` OpenStack network providing external connectivity.
+* `openshift_openstack_private_network_name` OpenStack network providing admin/control access for Ansible. It can be merged with other
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/configuration.md]: Done!

File: playbooks/openstack/configuration.md [hunk 8]
```diff
@@ -0,0 +1,463 @@
...
+right after provisioning will fail (unless you're using an external DNS server
+your provider network knows about). You must make sure your nodes are able to
+resolve each other by name.
+
+In `inventory/group_vars/all.yml`:
+
+* `openshift_openstack_provider_network_name` Provider network name. Setting this will cause the `openshift_openstack_external_network_name` and `openshift_openstack_private_network_name` parameters to be ignored.
+
+
+## Security Configuration
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/configuration.md]: Removed!

File: playbooks/openstack/configuration.md [hunk 9]
```diff
@@ -0,0 +1,463 @@
...
+running and the `openshift_hostname` variable must match the Nova server name
+for each node. The cluster deployment will fail without it. If you use the
+provided OpenStack dynamic inventory and configure the
+`openshift_openstack_dns_nameservers` Ansible variable, this will be handled
+for you.
+
+After a successful deployment, the cluster is configured for Cinder persistent
+volumes.
+
+### Validation
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/configuration.md]: Ah, okay - I'll probably iterate on this after the next round of updates for this PR.

File: playbooks/openstack/configuration.md [hunk 10]
```diff
@@ -0,0 +1,463 @@
...
+   * `openstack volume list` should show the volume attached to an OpenShift app node
+   * NOTE: this can take several seconds
+9. After a while, `oc get pod` should show the `mysql` pod as running
+10. Run `oc delete pod mysql` to remove the pod
+   * The Cinder volume should no longer be attached
+11. Run `oc delete pvc claim1` to remove the volume claim
+   * The Cinder volume should be deleted
+
+
+## Cinder-Backed Registry Configuration
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/openstack/configuration.md]: Ah, gotcha. I moved the existing registry documentation under this heading.

**Comment by tzumainn** [comment]
@tomassedovic Thanks for the review! I've updated the PR to take your suggestions into account, other than the cinder/DNS cleanup and openshift_master_default_subdomain clarification. I'll work on adding a few missing items, and then do the remaining cleanup.

**Comment by tzumainn** [comment]
@tomassedovic I've made the remaining changes (in separate commits so it's easier to spot them). I think that the DNS/cinder edits may best belong in a follow-up PR to separate out any potential discussion there. Let me know what you think!

File: playbooks/openstack/README.md [hunk 11]
```diff
@@ -202,52 +174,56 @@ $ ansible-playbook --user openshift \
...
+[configuration]: ./configuration.md
+[loadbalancer]: ./configuration.md#multi-master-configuration
+[external-dns]: ./configuration.md#dns-configuration
+[cinder-registry]: ./configuration.md#cinder-backed-registry-configuration
+[post-install]: ./post-install.md
+[configure-dns]: ./post-install.md#configure-dns
+[get-the-oc-client]: ./post-install.md#get-the-oc-client
+[log-in-using-the-command-line]: ./post-install.md#log-in-using-the-command-line
+[access-the-ui]: ./post-install.md#access-the-ui
+[scale-deployment]: ./post-install.md#scale-deployment
```
Comment (bogdando) [inline playbooks/openstack/README.md]: This refers to unsupported contrib repo. Let's please remove such references.

Comment (tzumainn) [reply playbooks/openstack/README.md]: Ah, missed this one - removed, thanks!

**Comment by tzumainn** [comment]
@bogdando Thanks for the review! I removed the link to the scaling actions. The only openshift-ansible-contrib links left are in the custom playbook section, and I think that's fair because it references playbooks that only exist in openshift-ansible-contrib right now. I checked the PRs you linked earlier, and I didn't see any references to 'advanced-configuration', but let me know if I missed something!

**Comment by tzumainn** [comment]
@tomassedovic I updated the DNS and cinder sections - those should be the last changes to this PR!

**Pull request merged**